yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -0,0 +1,264 @@
+            [g.Mark.O, g.Mark.X, g.Mark.X],
+            [g.Mark.X, g.Mark.O, g.Mark.X]]
+        self.assertEqual(game_state.board.cells, next_game_state_board)
+        self.assertEqual(game_state.step, 8)
+        self.assertEqual(game_state.round, 0)
+        self.assertEqual(game_state.phase, g.Phase.OUTROUND)
+        self.assertEqual(game_state.players[0].wins, 0)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+    def test_advance_next_round_action_outround(self):
+        previous_game_state_board = [
+            [g.Mark.X, g.Mark.O, g.Mark.X],
+            [None, g.Mark.X, g.Mark.O],
+            [g.Mark.X, None, g.Mark.O]]
+        game_state = StateSetup._set_state(
+            player1=StateSetup._set_player(g.Mark.X, 1),
+            board=g.Board(previous_game_state_board),
+            phase=g.Phase.OUTROUND, step=6)
+        act_queue_p1 = ListActionQueue([g.Action.new_start()])
+        g.Logic([act_queue_p1, []]).advance(game_state)
+        next_game_state_board = [[None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(game_state.board.cells, next_game_state_board)
+        self.assertEqual(game_state.step, 0)
+        self.assertEqual(game_state.round, 1)
+        self.assertEqual(game_state.phase, g.Phase.INROUND)
+        self.assertEqual(game_state.players[0].wins, 1)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+    def test_advance_next_round_action_onset(self):
+        game_state = StateSetup._set_state(phase=g.Phase.ONSET)
+        act_queue_p1 = ListActionQueue([g.Action.new_start()])
+        g.Logic([act_queue_p1, []]).advance(game_state)
+        next_game_state_board = [[None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(game_state.board.cells, next_game_state_board)
+        self.assertEqual(game_state.step, 0)
+        self.assertEqual(game_state.round, 0)
+        self.assertEqual(game_state.phase, g.Phase.INROUND)
+        self.assertEqual(game_state.players[0].wins, 0)
+        self.assertEqual(game_state.players[1].wins, 0)
+
+
+class LogicQueueActionTest(unittest.TestCase):
+    def test_win_cur_round_0(self):
+        game_state = StateSetup._set_state(phase=g.Phase.ONSET)
+        act_queue_p1 = ListActionQueue([
+            g.Action.new_start(),
+            g.Action.new_occupy(g.Cell(1, 1)),
+            None,
+            g.Action.new_occupy(g.Cell(0, 0)),
+            g.Action.new_occupy(g.Cell(0, 2)),
+            g.Action.new_occupy(g.Cell(2, 0))])
+        act_queue_p2 = ListActionQueue([
+            g.Action.new_occupy(g.Cell(1, 2)),
+            g.Action.new_occupy(g.Cell(2, 2)),
+            None,
+            g.Action.new_occupy(g.Cell(0, 1))])
+        finish_game_state_board = [
+            [g.Mark.X, g.Mark.O, g.Mark.X],
+            [None, g.Mark.X, g.Mark.O],
+            [g.Mark.X, None, g.Mark.O]]
+        while game_state.phase is not g.Phase.OUTROUND:
```

Comment:
Done.
Also I have fixed  `next()` method in `ListActionQueue` class.

Reply:
I think you should use `for i in range(0, len(act_queue_px.actions) + len(act_queue_po.actions))` (pay attention to whether the bounds are included), instead of using
```python
i = 0
while i <= len(act_queue_px.actions) + len(act_queue_po.actions):
    i += 1
```

Reply:
Done.